src/main/java/jenkins/branch/MultiBranchProject.java

```diff
@@ -161,6 +161,21 @@ protected MultiBranchProject(ItemGroup parent, String name) {
     public void onLoad(ItemGroup<? extends Item> parent, String name) throws IOException {
         super.onLoad(parent, name);
         init2();
+        // migrate any non-mangled names
+        ArrayList<P> items = new ArrayList<>(getItems());
+        for (P item : items) {
+            String itemName = item.getName();
+            String mangledName = NameMangler.apply(itemName);
+            if (!itemName.equals(mangledName)) {
+                if (getItem(mangledName) == null) {
+                    item.renameTo(mangledName);
+                    if (item.getDisplayNameOrNull() == null) {
+                        item.setDisplayName(itemName);
+                        item.save();
+                    }
+                }
```

Comment:
So the orphaned item strategy will remove them next time around as they will not have been indexed